src/callbacks_step/amr_dg2d.jl

```diff
@@ -96,13 +99,28 @@ function refine!(u_ode::AbstractVector, adaptor, mesh::Union{TreeMesh{2}, P4estM
     # Retain current solution data
     old_n_elements = nelements(dg, cache)
     old_u_ode = copy(u_ode)
-    GC.@preserve old_u_ode begin # OBS! If we don't GC.@preserve old_u_ode, it might be GC'ed
+    old_inverse_jacobian = copy(cache.elements.inverse_jacobian)
+    # OBS! If we don't GC.@preserve old_u_ode and old_inverse_jacobian, they might be GC'ed
+    GC.@preserve old_u_ode old_inverse_jacobian begin
         old_u = wrap_array(old_u_ode, mesh, equations, dg, cache)
 
+        if mesh isa P4estMesh
+            # Loop over all elements in old container and scale the old solution by the Jacobian
+            # prior to projection
+            for old_element_id in 1:old_n_elements
+                for v in eachvariable(equations)
+                    old_u[v, .., old_element_id] .= (old_u[v, .., old_element_id] ./
+                                                     old_inverse_jacobian[..,
+                                                                          old_element_id])
+                end
+            end
+        end
+
         reinitialize_containers!(mesh, equations, dg, cache)
 
         resize!(u_ode,
-                nvariables(equations) * nnodes(dg)^ndims(mesh) * nelements(dg, cache))
+                nvariables(equations) * nnodes(dg)^ndims(mesh) *
+                nelements(dg, cache))
```

Comment:
Out of curiosity: Is this due to the formatter or did you change this manually?

Reply:
It was from the formatter, but now that I have removed one of the loop blocks I can adjust it manually to put the `nelements` on the same line.

Reply:
:bulb:

Reply:
Fixed in 7761bfdf7787091015c5afca522d7d616df9edc3